src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -113,6 +113,17 @@ public Http2Stream(HttpRequestMessage request, Http2Connection connection, int s
                     _requestBodyCancellationSource = new CancellationTokenSource();
                 }
 
+                if (_request.HasHeaders && _request.Headers.ExpectContinue == true)
```

Comment:
I think this code should all be under the "else" statement above, so it only executes if _request.Content != null.

Reply:
updated. I was wondering about that but the original code did not have such test.